### PR TITLE
[NSX] Refactor API wrapper operations

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR:
- Centralices the API operations into the NsxApiClient class and decouples it from the NsxResource
- Adds helper methods to obtain the NSX element ids by the CloudStack entities names

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxGuestNetworkGuru.java [hunk 1]
```diff
@@ -265,20 +275,22 @@ public void commitMigration(NicProfile nic, Network network, VirtualMachineProfi
...
-            String vpcName = null;
-        if (nonNull(networkVO.getVpcId())) {
-            VpcVO vpc = _vpcDao.findById(networkVO.getVpcId());
-            if (isNull(vpc)) {
-                throw new CloudRuntimeException(String.format("Failed to find VPC network with id: %s", networkVO.getVpcId()));
-            }
-            vpcName = vpc.getName();
+        VpcVO vpc = _vpcDao.findById(networkVO.getVpcId());
+        if (isNull(vpc)) {
+            throw new CloudRuntimeException(String.format("Failed to find VPC network with id: %s", networkVO.getVpcId()));
```
Comment (Pearl1594) [inline plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxGuestNetworkGuru.java:280]: This refactor may not be required. The idea of it being the way it was previously written was that isolated networks could also be created as segments, and they would be connected to tier0

**Review by Pearl1594 — APPROVED** [review]
Looks good, left one comment

File: plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxGuestNetworkGuru.java (hunk 1 above)
Comment (nvazquez) [reply plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxGuestNetworkGuru.java]: Thanks, reverted that part

**Comment by Pearl1594** [comment]
@blueorangutan package

**Pull request merged**